### PR TITLE
[EP-18684-PearsonX] integration created initial

### PR DESCRIPTION
**Pull request by kiku1705**

### CODE REVIEW

File: app/service/pearsonx_integration.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  def get_content(_options = {})
+    begin
+      api_url = @credentials['host_url'] + PROGRAM_PATH
+      params = {
+        client_id: @credentials['client_id'],
+        client_secret: @credentials['client_secret']
+      }
+      params.merge!(timestamp: DateTime.parse(@credentials['last_polled_at']).to_i) if @credentials['last_polled_at']
+      programs = json_request(api_url, :get, params: params)
```
[the remark this answers is not in the document]
Comment (GaneshNavale) [reply app/service/pearsonx_integration.rb]:
```

File: app/service/pearsonx_integration.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      raise Webhook::Error::IntegrationFailure, "[pearsonxIntegration] Failed Integration for source #{@credentials['source_id']}, ErrorMessage: #{err.message}"
+    end
+  end
+
+  def content_item_attributes(entry)
+    {
+      external_id: entry['nid'],
+      source_id: @credentials['source_id'],
+      url: entry['link'],
+      name: sanitize_content(entry['title']),
```
[the remark this answers is not in the document]
Comment (kiku1705) [reply app/service/pearsonx_integration.rb]: Used to stip html tags from text.

[the remark this answers is not in the document]
Comment (GaneshNavale) [reply app/service/pearsonx_integration.rb]: but the title does not contain any HTML tags.

**Comment by kiku1705** [comment]
@GaneshNavale Please check updated the PR.

**Pull request merged**